src/source/src/connector_source.rs

```diff
@@ -113,21 +144,31 @@ impl InnerConnectorSourceReader {
                     break;
                 }
 
-                chunk = self.reader.next() => {
-                    match chunk.map_err(|e| internal_error(e.to_string())) {
-                        Err(e) => {
-                            log::error!("connector reader {} error happened {}", id, e.to_string());
-                            output.send(Either::Right(e)).await.ok();
-                            break;
-                        },
-                        Ok(None) => {
-                            log::warn!("connector reader {} stream stopped", id);
-                            break;
-                        },
-                        Ok(Some(msg)) => {
-                            output.send(Either::Left(msg)).await.ok();
-                        },
-                    }
+                c = self.reader.next() => {
+                    chunk = c;
+                }
+            }
+
+            match chunk.map_err(|e| internal_error(e.to_string())) {
+                Err(e) => {
+                    log::error!("connector reader {} error happened {}", id, e.to_string());
+                    output.send(Either::Right(e)).await.ok();
+                    break;
+                }
+                Ok(None) => {
+                    log::warn!("connector reader {} stream stopped", id);
+                    break;
+                }
+                Ok(Some(msg)) => {
+                    self.metrics
+                        .partition_input_count
+                        .with_label_values(&[
+                            self.context.actor_id.to_string().as_str(),
```

Comment:
To reduce overhead, a new field "actor_id_str" can be added to self.context and then used here. In this way, frequent type conversions can be avoided.